torchrl/data/replay_buffers/rb_prototype.py

```diff
@@ -0,0 +1,177 @@
+import collections
+from concurrent.futures import ThreadPoolExecutor
+from typing import Any, Callable, Optional, Sequence, Union, Tuple
+
+import torch
+
+from ..tensordict.tensordict import TensorDictBase, LazyStackedTensorDict
+from .replay_buffers import pin_memory_output, stack_tensors, stack_td
+from .samplers import Sampler, RandomSampler
+from .storages import Storage, ListStorage
+from .utils import INT_CLASSES, to_numpy
+from .writers import Writer, RoundRobinWriter
+
+
+class ReplayBuffer:
+    def __init__(
+        self,
+        storage: Storage = None,
+        sampler: Sampler = None,
+        writer: Writer = None,
+        collate_fn: Callable = None,
+        pin_memory: bool = False,
+        prefetch: int = None,
+    ) -> None:
+        self._storage = storage or ListStorage(size=1_000)
```

Comment:
It isn't enforced in the `ListStorage`, but it's enforced in the `Writer`. That should be enough - as all the writes are performed by the writer anyways - so the `size` will be respected, but I'm happy to add the enforcement logic to `ListStorage` itself as well, if you think there's a reason to do that.

Reply:
I'm a bit torn, but more in favour of enforcing.

I see why we would not enforce it (a list can be as long as we want, unlike a pre allocated array), but it happens that it gets longer than expected we should get a warning or an error. It would make sense to enforce it IMO

Reply:
+1 for enforcing it

Reply:
Ok - it's enforced now as part of `ListStorage.set` - I also renamed the `size` attribute to `max_size` as per https://github.com/facebookresearch/rl/pull/330#discussion_r941973165